project/apps/questions-service/src/questions.service.ts

```diff
@@ -1,14 +1,104 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+  CreateQuestionDto,
+  DeleteQuestionDto,
+  QuestionDto,
+  UpdateQuestionDto,
+} from '@repo/dtos/questions';
+import { createClient, SupabaseClient } from '@supabase/supabase-js';
 
 @Injectable()
 export class QuestionsService {
-  private readonly questions = [
-    { id: '1', title: 'What is NestJS?', userId: '1' },
-    // ...other questions
-  ];
-
-  findById(id: string) {
-    console.log('hello');
-    return this.questions.find((question) => question.id === id);
+  private supabase: SupabaseClient;
+  private readonly logger = new Logger(QuestionsService.name);
```

Comment:
i think for logger can consider using nestjs-pino with pino-pretty for best logging experience:
https://www.tomray.dev/nestjs-logging can read the second section on how he does it

Reply:
for `pino-pretty`, it prettifies http requests and responses, but for nestjs microservices don't use HTTP (right?) but TCP instead 
https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g30/blob/58dae0a3252eb0a1a5ed3d1ec47cba65047e3f19/project/apps/questions-service/src/main.ts#L9

Reply:
i think it would make more sense for the API gateway to use this logger to benefit from pino-pretty and pino-http, i can help u add that in if u want

Reply:
@javinchua 

Pino logger:
<img width="940" alt="Screenshot 2024-09-24 at 3 39 45 PM" src="https://github.com/user-attachments/assets/7528a0c1-4151-4ae0-9519-8aee35a164ec">

NestJS default logger:
<img width="996" alt="Screenshot 2024-09-24 at 3 40 45 PM" src="https://github.com/user-attachments/assets/b50f4345-0884-4131-9272-7281eb7bc083">

Reply:
nice! i assume this is with pino-pretty? whats ur thoughts